File: 백준/Bronze/1085. 직사각형에서 탈출/직사각형에서 탈출.py
x, y, w, h = map(int,input().split(' '))
min_x = 0
min_y = 0
if (w / 2) >= x:
    min_x = x
elif (w / 2) < x:
    min_x = w-x
if (h / 2) >= y:
    min_y = y
elif (h / 2) < y:
    min_y = h - y
if min_x < min_y:
    print(min_x)
else:
    print(min_y)

File: 백준/Bronze/1085. 직사각형에서 탈출/직사각형에서 탈출.c
#include <stdio.h>
int f_min(c_p,len) {
	if (len / 2 >= c_p) {
		return c_p;
	}
	else {
		return len - c_p;
	}
}

int main(void) {
	int x, y, w, h;
	int min_x = 0;
	int min_y = 0;
	scanf("%d %d %d %d", &x, &y, &w, &h);
	min_x = f_min(x, w);
	min_y = f_min(y, h);
	if (min_x > min_y) {
		printf("%d", min_y);
	}
	else {
		printf("%d", min_x);
	}

	return 0;
}